### ISSUE TITLE
Add `edition = "2018"` to the Cargo.toml

### ISSUE THREAD

**Issue by hfiguiere** [gtk-rs/gir#1039]
When updating the crate, either -sys or not, with gir, gir should make sure `edition = "2018"` is in the Cargo.toml, otherwise the crate won't build.

**Comment by sdroege** [comment]
Good idea, do you want to make a PR?

gir is only touching the -sys `Cargo.toml` though, never the non-sys one. So that would require some new code.

**Comment by hfiguiere** [comment]
I'll see what I can do. Indeed the non-sys Cargo isn't changed.

**Comment by gabbyprecious** [comment]
@bilelmoussaoui would love to also take on this

**Comment by ps-19** [comment]
@hfiguiere sir i have a question. if edition="2022" is written (more precisely used) is it better or need to change.

**Comment by ps-19** [comment]
@sdroege @bilelmoussaoui i need feedback.

**Comment by bilelmoussaoui** [comment]
It should actually use edition=2022 nowadays.

**Comment by ps-19** [comment]
is this issue is still relevent then?

**Comment by ps-19** [comment]
> It should actually use edition=2022 nowadays.

Thank you for you response.

**Comment by bilelmoussaoui** [comment]
> is this issue is still relevent then?

Yes it is. Currently the generated Cargo.toml doesn't contain any `edition` specified.

**Comment by ps-19** [comment]
https://github.com/gtk-rs/gir/blob/master/Cargo.toml#L11 it is containing here.

**Comment by bilelmoussaoui** [comment]
That is for the gir project, not for the projects generated with gir itself which is what this issue is about

**Comment by ps-19** [comment]
@bilelmoussaoui sir can you please describe me the steps required to reproduce the issue.

**Comment by bilelmoussaoui** [comment]
Please have a look at how to generate the bindings using gir first. E.g https://gtk-rs.org/gir/book/tutorial/sys_library.html

**Comment by hfiguiere** [comment]
> It should actually use edition=2022 nowadays.

There is a 2022 edition?
I know that Debian Bullseye still doesn't recognize 2021 with the distro provided rust toolchain...

**Comment by bilelmoussaoui** [comment]
Oh sorry, it is 2021.

**Comment by Semeyian** [comment]
Hello @bilelmoussaoui,
I am Mercy Semeyian from the Outreachy program. I've been going through the tutorial you provided above on how to generate the bindings using Gir. The `gir-files` directory doesn't contain a `GtkSource-3.0.gir` file for sourceview. Could it have been replaced with `Git-3.0.gir`, or is just missing entirely? If so, how should I proceed to create the `Gir.toml` `library` option?
Thank you. :)

**Comment by A6GibKm** [comment]
Is this resolved? See e.g. https://github.com/gtk-rs/gtk4-rs/commit/7602a3b7ffccaaf9cccdce7719cb1e2c67d67feb and https://github.com/gtk-rs/gir/commit/364678bf8674d8bdda02fde783f639abd000d245.

**Comment by bilelmoussaoui** [comment]
I think using workspaces is better, yes.
